src/library/dao/autodjcratesdao.cpp

```diff
@@ -881,3 +886,81 @@ void AutoDJCratesDAO::slotPlayerInfoTrackUnloaded(QString group,
         }
     }
 }
+//
+// We are selecting the track in the following manner.
+// We divide the library tracks into two section . For which
+// we sort the library according to times_played and select a
+// percentage_of_prefered_tracks (25% by default) .
+//
+// What we do then is either select a random track from this %25
+// or a random track from the remaining 75% both with equal
+// probability .
+//
+// (Assuming that 25% tracks are unplayed or very rarely played)
+//
+// This ensures that there is a 50% chance that an rarely played
+// track is returned .
+//
+// This is better than returning the least played track because
+// in that case there may be instances where the same track is
+// returned by this function on multiple successive calls which is
+// not desired.
+//
+// Further this also does not restrict our function to only retrieve
+// unplayed tracks (there is probably a reason they are unplayed).
+///
+int AutoDJCratesDAO::getRandomTrackIdFromLibrary(void) {
```

Comment:
> .. there is probably a reason they are unplayed

Yes, this is a very good point. This feature has all tracks including the crap of intros autos and other tracks we don't want to hear.